js/src/forum/components/ReplyComposer.js

```diff
@@ -68,10 +58,16 @@ export default class ReplyComposer extends ComposerBody {
    * @return {Object}
    */
   data() {
-    return {
-      content: this.content(),
-      relationships: { discussion: this.props.discussion },
-    };
+    const data = super.data();
+    data.relationships = { discussion: this.props.discussion };
```

Comment:
What if `data.relationships` isn't an empty object? You are deleting relationship data then.

Reply:
Not necessarily. super.data() returns a new object, not a saved one. We could do `const data = {...super.data()}` instead though to be totally safe in case of future changes.

Reply:
If something extends the Composer `data` method, and returns relationship data there, you are getting rid of it. That's what I mean.

Reply:
That seems to be the default current behavior though? We could try to merge the objects with `Object.assign` perhaps?

Reply:
I don't think Object.assign works that way. The current behavior doesn't have a parent data model that you are calling, so if you start to call it, you should use everything it returns _OR_ make it known that the relationship data is going to be thrown out. This could cause a lot of wasted time on debugging if you just replace the relationships.

Reply:
Would merging the relationships objects be good?

Reply:
Hm, perhaps assigning data.relationships would work. Or `if (!data.relationships) data.relationships = {}`.

Reply:
Improved, is this better?